### ISSUE TITLE
Provide a cjs wrapper built in to node-fetch

### ISSUE THREAD

**Issue by jimmywarting** [node-fetch/node-fetch#1327]
Based on the documentation jimmywarting linked above, I created a helper typescript file that hopefully can serve other typescript users that are stuck:

```ts
// node-fetch.ts
// Helps importing node-fetch based on https://github.com/node-fetch/node-fetch/issues/1279

import type fetch from 'node-fetch';
export { RequestInit } from 'node-fetch'; // TODO: add more exports as needed here


// eslint-disable-next-line no-eval
const fetchPromise: Promise<typeof fetch> = eval('import("node-fetch")').then((mod: { default: typeof fetch }) => mod.default);
const exportedFetch: typeof fetch = (...args) => fetchPromise.then(fetch => fetch(...args));
export default exportedFetch;
```
Now, just import it with `import fetch, { RequestInit } from './node-fetch';` almost as usual (note the `./`)

It would be handy if such script was directly embedded in the package, like `import fetch, { RequestInit } from 'node-fetch/cjs';`.

_Originally posted by @jeremyVignelles in https://github.com/node-fetch/node-fetch/issues/1266#issuecomment-925813598_

**Comment by jeremyVignelles** [comment]
The eval method is an acceptable for me and would solve the friction users have upgrading to 3.x.

The only thing a developer using typescript would need to do is to replace
```ts
import fetch, { RequestInit } from 'node-fetch';
```

with
```ts
import fetch, { RequestInit } from 'node-fetch/cjs';
```

**Comment by jimmywarting** [comment]
@jeremyVignelles how about this instead?
```ts
import type { RequestInit, Headers, Request, Response } from 'node-fetch';
import fetch from 'node-fetch/cjs';
```

**Comment by jeremyVignelles** [comment]
OK for me, though I'd prefer to keep a single import per dependency. The more dependency you have in a project, the longer the section becomes and it can be a nightmare to read...

**Comment by jimmywarting** [comment]
I'm also wondering about the naming if `cjs` is a good name for it. it can imply that it exports the same thing but with `require` instead... perhaps something like `node-fetch/cjs-wrapper` or something else would be better?

**Comment by jeremyVignelles** [comment]
> it can imply that it exports the same thing 
It does export the same thing, that can be used the same way. The way it's done internally is an implementation detail.

> but with `require` instead
As a consumer, I'd tend to interpret this as a "I can import this with require()" (We just need to make sure that it actually can :smile: ). As a consumer not knowing the difference between cjs and mjs, I'd just do what I'm told in the README anyway... That said, I'm ok with `cjs-wrapper` too

**Comment by jimmywarting** [comment]
> It does export the same thing

Well, yes but only the `fetch` method... not the rest

**Comment by jeremyVignelles** [comment]
Indeed, the classes cannot be imported that way synchronously...

What about something like:
```ts
export const RequestPromise: typeof Request = fetchPromise.then(module => module.Request);
```

imported like:
```
import fetch, { RequestPromise } from 'node-fetch/cjs';

// Later
const Request = await RequestPromise;
```

The usage differs a lot now. Not sure if there's a better workaround available...

**Comment by jimmywarting** [comment]
```ts
// Later
const Request = await RequestPromise;
```
cjs can't use top level await... if they do use `await` then they can just as well use `everything = await import('node-fetch')`

**Comment by jeremyVignelles** [comment]
That's what I meant by `later`, it's inside an async function...

> they can just as well use `everything = await import('node-fetch')`

Except for typescript where it's translated into a `require()` call, which is what we want to avoid with this wrapper...

There should be a way to access the classes, but other than the workaround I suggested (using the `Promise<Request>`), I don't see any other workaround... Not sure it does even work...

**Comment by Donnevtis** [comment]
Some trouble. It's a mess. I can't use ES modules in a serverless environment. The typescript compiles a dynamic import to `require`. `@types/node-fetch` is deprecated. What do you want me to do?

**Comment by LinusU** [comment]
> Except for typescript where it's translated into a require() call, which is what we want to avoid with this wrapper...

This is fixed in the latest TypeScript beta, which should be out in ~1 month 🎉 

https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/

**Comment by jimmywarting** [comment]
still wondering about what we should be doing about this issue, would like to hear someones tough if this should be done before working on v4

**Comment by jeremyVignelles** [comment]
The fact that typescript postponed the feature is disappointing, and we're stuck in the same mess as before.

I'm not a fan of needing an await for importing classes, but user-made workarounds will have the same issue, and the classes are used in few use cases, right?

I'd embed the workaround with proper warnings and in a separate file. Users must opt-in for the workaround until it is fixed by typescript.

What else could we do?
